exo-go/src/docker_setup/index.go

```diff
@@ -0,0 +1,138 @@
+package dockerSetup
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Originate/exosphere/exo-go/src/app_dependency_helpers"
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/os_helpers"
+	"github.com/Originate/exosphere/exo-go/src/service_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	"github.com/Originate/exosphere/exo-go/src/util"
+	"github.com/pkg/errors"
+)
+
+// DockerSetup renders docker-compose.yml file with service configuration
+type DockerSetup struct {
+	AppConfig     types.AppConfig
+	ServiceConfig types.ServiceConfig
+	ServiceData   types.ServiceData
+	Role          string
+	Logger        *logger.Logger
+	AppDir        string
+}
+
+// NewDockerSetup is DockerSetup's constructor
+func NewDockerSetup(appConfig types.AppConfig, serviceConfig types.ServiceConfig, serviceData types.ServiceData, role string, logger *logger.Logger, appDir string) *DockerSetup {
+	return &DockerSetup{
+		AppConfig:     appConfig,
+		ServiceConfig: serviceConfig,
+		ServiceData:   serviceData,
+		Role:          role,
+		Logger:        logger,
+		AppDir:        appDir,
+	}
+}
+
+func (dockerSetup *DockerSetup) getDockerEnvVars() map[string]string {
+	result := map[string]string{"ROLE": dockerSetup.Role}
+	for _, dependency := range dockerSetup.AppConfig.Dependencies {
+		builtDependency := appDependencyHelpers.Build(dependency, dockerSetup.AppConfig, dockerSetup.AppDir)
+		for variable, value := range builtDependency.GetServiceEnvVariables() {
+			result[variable] = value
+		}
+	}
+	for _, dependency := range dockerSetup.ServiceConfig.Dependencies {
+		result[strings.ToUpper(dependency.Name)] = dependency.Name
+	}
+	return result
+}
+
+func (dockerSetup *DockerSetup) getDockerLinks() []string {
+	result := []string{}
+	for _, dependency := range dockerSetup.ServiceConfig.Dependencies {
+		result = append(result, fmt.Sprintf("%s%s:%s", dependency.Name, dependency.Version, dependency.Name))
```

Comment:
Hmm, If thats whats being used I think that would be a good thing in case we want to change the container names down the road.